*** newlib-1.17.0/newlib/libc/stdio/refill.c.orig	2010-06-11 08:01:56.206278608 -0500
--- newlib-1.17.0/newlib/libc/stdio/refill.c	2010-06-11 07:57:52.806293255 -0500
***************
*** 104,110 ****
--- 104,114 ----
     */
  
    if (fp->_flags & (__SLBF | __SNBF))
+   {
      _CAST_VOID _fwalk (_GLOBAL_REENT, lflush);
+     if (_GLOBAL_REENT != ptr)
+       _CAST_VOID _fwalk (ptr, lflush);
+   }
    fp->_p = fp->_bf._base;
    fp->_r = fp->_read (ptr, fp->_cookie, (char *) fp->_p, fp->_bf._size);
  #ifndef __CYGWIN__
