*** cpukit/libnetworking/rtems/rtems_mii_ioctl_kern.c.orig	2018-08-08 08:36:17.000000002 -0700
--- cpukit/libnetworking/rtems/rtems_mii_ioctl_kern.c	2018-08-08 08:36:37.000000002 -0700
***************
*** 118,124 ****
  
      /* link status */
      if (BMSR_LINK & bmsr)
!       options |= IFM_LINK_OK | IFM_ACTIVE | IFM_AVALID;
  
      /* do we have autonegotiation disabled ? */
      if (!(BMCR_AUTOEN & bmcr)) {
--- 118,124 ----
  
      /* link status */
      if (BMSR_LINK & bmsr)
!       options |= IFM_LINK_OK;
  
      /* do we have autonegotiation disabled ? */
      if (!(BMCR_AUTOEN & bmcr)) {
